db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java

```diff
@@ -3660,23 +3673,19 @@ public void setupSubStatementSessionContext(Activation a) throws StandardExcepti
     }
 
     @Override
-    public SQLSessionContext getTopLevelSQLSessionContext() {
+    public SQLSessionContext getTopLevelSQLSessionContext() throws StandardException {
         if(topLevelSSC==null){
-            topLevelSSC=new SQLSessionContextImpl(
-                    getInitialDefaultSchemaDescriptor(),
-                    getSessionUserId(),
-                    defaultRoles, groupuserlist);
+            topLevelSSC = createSQLSessionContext();
         }
         return topLevelSSC;
     }
 
 
     @Override
-    public SQLSessionContext createSQLSessionContext() {
-        return new SQLSessionContextImpl(
-                getInitialDefaultSchemaDescriptor(),
-                getSessionUserId() /* a priori */,
-                defaultRoles, groupuserlist);
+    public SQLSessionContext createSQLSessionContext() throws StandardException {
+        return new SQLSessionContextImpl(getDataDictionary().getDatabaseDescriptor(dbName, getTransactionCompile()),
```

Comment:
Would it be possible to cache the DatabaseDescriptor in the `initialize()` method similarly to what we do with the SchemaDescriptor, so that we don't have to throw StandardExceptions all over the place?

Reply:
Done! That's a lot simpler now, thanks!